### PR TITLE
Double ended queue java

### PR DESCRIPTION
**Pull request by khusheekapoor**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes, I have read the Contributing Guidelines on Pull Requests

### Description

I have written a code to implement basic operations (pushfront, pushrear, popfront, poprear, display) on Double Ended Queues in Java as a GSSoC'21 participant.

### Checklist

- [Yes] I've read the contribution guidelines.
- [Yes] I've checked the issue list before deciding what to submit.
- [Yes] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

For Issue #4984 .

### CODE REVIEW

File: Java/ds/DEQueue.java [hunk 1]
```diff
@@ -0,0 +1,303 @@
+
```
Comment (ankitaggarwal23) [inline Java/ds/DEQueue.java]: Remove the space

File: Java/ds/DEQueue.java [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+ * to initialize the global parameters:
+ * integer array dequeue[] - to store the elements
+ * integer variable elements - to keep a count of the number of elements in the queue
+ * integer variable front - to point at the front end of the dequeue
+ * integer variable rear - to point at the rear end of the dequeue
+ * 
+ * To take input from the user, Scanner class has been instantiated.
+ * 
+ */
+import java.util.*;
```
Comment (ankitaggarwal23) [inline Java/ds/DEQueue.java:31]: Add a new line after header file

File: Java/ds/DEQueue.java [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+         * 1. Check if the queue is full (no. of elements==size).
+         *      If true, queue is full, print "Overflow" and exit.
+         *      If false, queue is not full, go to Step 2.
+         * 2. Check if there is space in the front end of the queue (front==0).
+         *      If true, front end is full, print "Front full" and exit.
+         *      If false, front end is not full, go to Step 3.
+         * 3. Increment the number of elements by 1.
+         * 4. Decrement the front pointer by 1.
+         * 5. Assign the memory space pointed by front the value received as formal parameter.
+         */
```
Comment (ankitaggarwal23) [inline Java/ds/DEQueue.java]: Remove the algo

File: Java/ds/DEQueue.java [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+        /* Algorithm:
+         * 1. Check if the queue is full (no. of elements==size).
+         *      If true, queue is full, print "Overflow" and exit.
+         *      If false, queue is not full, go to Step 2.
+         * 2. Increment the number of elements by 1.
+         * 3. Increment the rear pointer by 1.
+         * 4. Check if queue is empty (front==-1).
+         *      If true, change the pointer to point at the first element.
+         * 5. Assign the memory space pointed by front the value received as formal parameter.
+         */
```
Comment (ankitaggarwal23) [inline Java/ds/DEQueue.java]: Remove the algo

File: Java/ds/DEQueue.java [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+         * 1. Check if the queue is empty (no. of elements==0).
+         *      If true, queue is empty, print "Underflow" and exit.
+         *      If false, queue is not empty, go to Step 2.
+         * 2. Assignt a temporary variable the value pinted by the front pointer.
+         * 3. Increment the front pointer by 1.
+         * 3. Decrement the number of elements by 1.
+         * 4. Check if queue is empty (no. of elements==0).
+         *      If true, reset the front and rear pointers to original value.
+         * 5. Return the value stored in temporary variable.
+         */
```
Comment (ankitaggarwal23) [inline Java/ds/DEQueue.java]: Remove the algo

File: Java/ds/DEQueue.java [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+         * 1. Check if the queue is empty (no. of elements==0).
+         *      If true, queue is empty, print "Underflow" and exit.
+         *      If false, queue is not empty, go to Step 2.
+         * 2. Assignt a temporary variable the value pinted by the rear pointer.
+         * 3. Decrement the rear pointer by 1.
+         * 3. Decrement the number of elements by 1.
+         * 4. Check if queue is empty (no. of elements==0).
+         *      If true, reset the front and rear pointers to original value.
+         * 5. Return the value stored in temporary variable.
+         */
```
Comment (ankitaggarwal23) [inline Java/ds/DEQueue.java]: Remove the algo

File: Java/ds/DEQueue.java [hunk 7]
```diff
@@ -0,0 +1,303 @@
...
+            System.out.println();
+        }
+    }
+
+    void main()
+    {
+        // Driver code to call all the methods, until the user wishes
+
+        System.out.println("Enter size of the dequeue: ");
+        int size = scanner.nextInt(); scanner.nextLine();
```
Comment (ankitaggarwal23) [inline Java/ds/DEQueue.java]: Shift ` scanner.nextLine();` to new line

File: Java/ds/DEQueue.java [hunk 8]
```diff
@@ -0,0 +1,303 @@
...
+        System.out.println("1. Inserting element in the front of the dequeue");
+        System.out.println("2. Inserting element in the rear of the dequeue");
+        System.out.println("3. Deleting element from the front of the dequeue");
+        System.out.println("4. Deleting element from the rear of the dequeue");
+        System.out.println("5. Displaying the dequeue");
+
+        while(flag!=0)
+        {
+            System.out.println("Enter your choice");
+            int choice = scanner.nextInt(); scanner.nextLine();
```
Comment (ankitaggarwal23) [inline Java/ds/DEQueue.java]: Shift ` scanner.nextLine();` to new line

File: Java/ds/DEQueue.java [hunk 9]
```diff
@@ -0,0 +1,303 @@
...
+            switch(choice)
+            {
+                case 1:
+                System.out.println("Enter element to be inserted: ");
+                value = scanner.nextInt(); scanner.nextLine();
+                ob.pushfront(value);
```
Comment (ankitaggarwal23) [inline Java/ds/DEQueue.java]: Give a one tab space

File: Java/ds/DEQueue.java [hunk 10]
```diff
@@ -0,0 +1,303 @@
...
+                break;
+
+                case 2:
+                System.out.println("Enter element to be inserted: ");
+                value = scanner.nextInt(); scanner.nextLine();
+                ob.pushrear(value);
```
Comment (ankitaggarwal23) [inline Java/ds/DEQueue.java]: Give a one tab space

File: Java/ds/DEQueue.java [hunk 11]
```diff
@@ -0,0 +1,303 @@
...
+                break;
+
+                case 3:
+                ob.popfront();
```
Comment (ankitaggarwal23) [inline Java/ds/DEQueue.java]: Give a one tab space

File: Java/ds/DEQueue.java [hunk 12]
```diff
@@ -0,0 +1,303 @@
...
+                break;
+
+                case 4:
+                ob.poprear();
```
Comment (ankitaggarwal23) [inline Java/ds/DEQueue.java]: Give a one tab space

File: Java/ds/DEQueue.java [hunk 13]
```diff
@@ -0,0 +1,303 @@
...
+                break;
+
+                case 5:
+                ob.display();
```
Comment (ankitaggarwal23) [inline Java/ds/DEQueue.java]: Give a one tab space

File: Java/ds/DEQueue.java [hunk 14]
```diff
@@ -0,0 +1,303 @@
...
+                break;
+            }
+            System.out.println("Would you like to continue? Enter 1 for YES and 0 for NO:");
+            flag = scanner.nextInt(); scanner.nextLine();
```
Comment (ankitaggarwal23) [inline Java/ds/DEQueue.java]: Shift scanner.nextLine(); to new line

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by khusheekapoor** [comment]
Made the changes @ankitaggarwal23 . Let me know if any further changes are required. Thanks.

**Review by ankitaggarwal23 — APPROVED** [review]

**Comment by HarshCasper** [comment]
Please make an issue here and raise a PR, for an article as part of our Docs repository: https://github.com/TesseractCoding/NeoAlgo-Docs

It would be much appreciated. Thanks!

**Pull request merged**